tutorials/asr/Streaming_ASR.ipynb

```diff
@@ -0,0 +1,699 @@
+    "\n",
+    "An audio buffer is made up of a chunk of audio with some padded audio from previous chunk to provide context for the beginning and ending portions of the chunk. \n",
+    "\n",
+    "Let's suppose tha the maximum length of audio that can be transcribed with conformer-large model is 20s, then we can use 20s as the buffer size and use 15s (for example) as the chunk size, so the one hour audio is broken into 240 chunks of 15s each. Let's take a look at a few audio buffers that may be created for this audio."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# A simple iterator class to return successive chunks of samples\n",
+    "class AudioChunkIterator():\n",
+    "    def __init__(self, samples, frame_len, sample_rate):\n",
+    "        self._samples = samples\n",
+    "        self._chunk_len = chunk_len_in_secs*sample_rate\n",
+    "        self._start = 0\n",
+    "        self.output=True\n",
+    "   \n",
+    "    def __iter__(self):\n",
+    "        return self\n",
+    "    \n",
+    "    def __next__(self):\n",
+    "        if not self.output:\n",
+    "            raise StopIteration\n",
+    "        last = int(self._start + self._chunk_len)\n",
+    "        if last <= len(self._samples):\n",
+    "            chunk = self._samples[self._start: last]\n",
+    "            self._start = last\n",
+    "        else:\n",
+    "            chunk = np.zeros([int(self._chunk_len)], dtype='float32')\n",
+    "            samp_len = len(self._samples) - self._start\n",
+    "            chunk[0:samp_len] = self._samples[self._start:len(self._samples)]\n",
+    "            self.output = False\n",
+    "   \n",
+    "        return chunk"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# a helper function for extracting samples as a numpy array from the audio file\n",
+    "import soundfile as sf\n",
+    "def get_samples(audio_file, target_sr=16000):\n",
+    "    with sf.SoundFile(audio_file, 'r') as f:\n",
+    "        dtype = 'int16'\n",
+    "        sample_rate = f.samplerate\n",
+    "        samples = f.read(dtype=dtype)\n",
+    "        if sample_rate != target_sr:\n",
+    "            samples = librosa.core.resample(samples, sample_rate, target_sr)\n",
+    "        samples=samples.astype('float32')/32768\n",
+    "        samples = samples.transpose()\n",
+    "        return samples"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's take a look at each chunk of speech that is used for decoding."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import matplotlib.pyplot as plt\n",
+    "samples = get_samples(repeated_audio_path)\n",
+    "SAMPLE_RATE=16000\n",
```

Comment:
Maybe get this from the model rather than hardcode? Its fine either way since we mostly train only for 16kHz